src/MockCollection.php

```diff
@@ -80,24 +87,60 @@ public function updateOne($filter, $update, array $options = [])
         $matcher = $this->matcherFromQuery($filter);
         foreach ($this->documents as $i => &$doc) {
             if ($matcher($doc)) {
-                foreach ($update['$set'] ?? [] as $k => $v) {
-                    $doc[$k] = $v;
-                }
+                $this->updateCore($doc,$update);
                 return;
             }
         }
+        $this->updateUpsert($filter,$update,$options,false);
     }
 
     public function updateMany($filter, $update, array $options = [])
     {
         $matcher = $this->matcherFromQuery($filter);
+        $anyUpdates = false;
         foreach ($this->documents as $i => &$doc) {
             if (!$matcher($doc)) {
                 continue;
             }
 
-            foreach ($update['$set'] ?? [] as $k => $v) {
-                $doc[$k] = $v;
+            $this->updateCore($doc,$update); 
+            $anyUpdates = $anyUpdates || true;
```

Comment:
yea this is just an artifact of the fixup. I had originally written `$anyUpdates = $anyUpdates || !!$update['$set']`, but then from the docs I realized that in case of `!!$update['$set']`, the element just gets completely replaced (check [line 128](https://github.com/martin-helmich/php-mongomock/pull/1/files#diff-b0c966a257e74dfc0b81dff64b6cf9adR128))